distribution/test/itests/test-itests-common/src/main/java/org/codice/ddf/itests/common/config/UrlResourceReaderConfigurator.java

```diff
@@ -30,28 +31,63 @@
 
   private static final String PID = "ddf.catalog.resource.impl.URLResourceReader";
 
+  private static final String ROOT_RESOURCE_DIRECTORIES_PROPERTY_KEY = "rootResourceDirectories";
+
   private ConfigurationAdmin configAdmin;
 
   public UrlResourceReaderConfigurator(ConfigurationAdmin configAdmin) {
     this.configAdmin = configAdmin;
   }
 
   public void setUrlResourceReaderRootDirs(String... rootResourceDirs) throws IOException {
+    Set<String> newRootResourceDirs = ImmutableSet.<String>builder().add(rootResourceDirs).build();
+
     Configuration configuration = configAdmin.getConfiguration(PID, null);
+
+    if (configuration == null) {
+      LOGGER.warn(
+          "{} configuration was null, cannot update {}",
+          PID,
+          ROOT_RESOURCE_DIRECTORIES_PROPERTY_KEY);
+      return;
+    }
+
+    if (configuration.getProperties() != null) {
+      Set<String> currentRootResourceDirs =
+          ImmutableSet.copyOf(
+              (Collection<String>)
+                  configuration.getProperties().get(ROOT_RESOURCE_DIRECTORIES_PROPERTY_KEY));
```

Comment:
❓ Should we protect ourselves against the property not existing? Do we really need to wrap it in an immutable set for comparison? Maybe you can leave it up to the `rootResourceDirsEqual` method to account for `null` in case it did not exist and perform the comparison. You might be able to reuse that later when waiting